equinox/_misc.py

```diff
@@ -18,3 +21,17 @@ def default_floating_dtype():
         return jnp.float64
     else:
         return jnp.float32
+
+
+def default_init(
+    key: PRNGKeyArray, shape: tuple[int, ...], dtype: Any, lim: float
+) -> jax.Array:
+    if jnp.issubdtype(dtype, jnp.complexfloating):
+        # only two possible complex dtypes, jnp.complex64 or jnp.complex128
+        real_dtype = jnp.float32 if dtype == jnp.complex64 else jnp.float64
```

Comment:
I think a neater trick is this: https://github.com/patrick-kidger/lineax/blob/1909d190c1963d5f2d991508c1b2714f2266048b/lineax/_misc.py#L92-L93